File: spoj/blueprint/3.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int t;
    cin>>t;
    while(t--){
        int n;cin>>n;
        vector<int>arr;{for(int i=0;i<n;i++){int x;cin>>x;arr.push_back(x);}}
        sort(arr.begin(),arr.end());
        vector<int>fre(n+1,0);
        for(int val:arr)fre[val]++;
        int ind,maxi=0;
        for(int i=0;i<=n;i++){
            if(fre[i] > maxi){
                maxi = fre[i];
                ind = i;
            }
        }
        long  ans = (fre[ind]*(fre[ind]-1))/2;
        vector<int> fre1(n+1,0);
        arr.erase(unique(arr.begin(),arr.end()),arr.end());
        for(int val:arr)if(val != ind){
            ans += (fre[val]*(fre[val]-1))/2;
            fre1[abs(val - ind)]+= (fre[val])*fre[ind];
        }
        long temp = *max_element(fre1.begin(),fre1.end());
        cout<<max(ans,temp);
        cout<<" "<<ind<<endl;
    }
}
        
        

File: spoj/codeforces/ed90/3.py
for _ in range(int(input())):
    ss = input();
    arr = [0]*len(ss)
    val = 0
    ind = 0
    for ch in ss:
        if ch == '+':
            val += 1
        else:
            val -= 1
        arr[ind] = val
        ind += 1
    ans = 0
    mn = 0
#     print(arr)
    for ind,val in enumerate(arr):
        if val < mn:
            ans += ind+1
            mn = val
#             print(ans)
    ans += len(ss)
    print(ans)
               
